### PR TITLE
[2/n] upgrade react-native 0.76  

### PR DESCRIPTION
**Pull request by gabrieldonadel**
# Why

Follow up of https://github.com/expo/expo/pull/31552
Closes [ENG-13531](https://linear.app/expo/issue/ENG-13531)

# How

- Bump expo go use react-native 0.76.0-rc.4 based on the upgrade helper
- Bump our react-native fork to  0.76.0-rc.4

# Test Plan

- Expo Go on Android and iOS

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: apps/expo-go/android/settings.gradle [hunk 1]
```diff
@@ -9,7 +9,21 @@ pluginManagement {
   }
 }
 plugins { id("com.facebook.react.settings") }
-extensions.configure(com.facebook.react.ReactSettingsExtension){ ex -> ex.autolinkLibrariesFromCommand() }
+
+extensions.configure(com.facebook.react.ReactSettingsExtension) { ex ->
+  println('\u001B[32mUsing expo-modules-autolinking as core autolinking source\u001B[0m')
```
Comment (Kudo) [inline apps/expo-go/android/settings.gradle:14]:
```suggestion
```

File: apps/expo-go/ios/Exponent/Versioned/Core/EXVersionManagerObjC.mm [hunk 2]
```diff
@@ -45,7 +45,7 @@
 #endif
 
 // Import 3rd party modules that need to be scoped.
-#import <RNCAsyncStorage/RNCAsyncStorage.h>
+#import "RNCAsyncStorage/RNCAsyncStorage.h"
```
Comment (Kudo) [inline apps/expo-go/ios/Exponent/Versioned/Core/EXVersionManagerObjC.mm:48]:
```suggestion
#import <RNCAsyncStorage/RNCAsyncStorage.h>
```

nit: for clang it makes more sense to have import <...> format then double quoted. if that don't break, i would like to keep the original code.

File: apps/expo-go/ios/Podfile [hunk 3]
```diff
@@ -51,8 +51,19 @@ target 'Expo Go' do
   # Install vendored pods.
   use_pods! 'vendored/unversioned/**/*.podspec.json'
 
-  # React Native
-  use_native_modules!
+  Pod::UI.puts('Using expo-modules-autolinking as core autolinking source'.green)
```
Comment (Kudo) [inline apps/expo-go/ios/Podfile:54]:
```suggestion
```

**Review by lukmccall — APPROVED** [review]

**Review by Kudo — APPROVED** [review]
also please help to merge this pr before ci all green. thanks!

**Pull request merged**